src/main/kotlin/org/jetbrains/research/libsl/nodes/references/TypeReference.kt

```diff
@@ -11,11 +11,20 @@ open class TypeReference(
     override val context: LslContextBase
 ) : LslReference<Type, TypeReference> {
     override fun resolve(): Type? {
-        return resolveArrayType() ?:
-        resolveListType() ?:
-        resolveMapType() ?:
-        resolveNullType() ?:
-        context.resolveType(this)
+        return resolveArrayType() ?: resolveListType() ?: resolveMapType() ?: resolveNullType() ?: resolveLiteralType()
+        ?: context.resolveType(this)
+    }
+
+    private fun resolveLiteralType(): Type? {
+        // TODO: add more complicated checkers;
+        return when {
+            name.first() == '\"' -> StringType(context)
+            name[0] == '\'' -> CharType(context)
+            ((Character.isDigit(name.first()) || (name[0] == '-' && Character.isDigit(name[1]))) && name.contains(".")) -> Float64Type(context)
+            Character.isDigit(name.first()) -> Int64Type(context)
+            (name == "false" || name == "true") -> BoolType(context)
+            else -> null
+        }
```

Comment:
Can this information be resolved during creation of this object and stored within the instance?
I.e. when we parse a literal string - store `StringType(context)` as an additional field (like `literalType: Type?`).
This can also be used to create a helper method `isLiteral(): bool = literalType != null`.

Reply:
Fixed;